tools/sense_studio/train_module.py

```diff
@@ -0,0 +1,84 @@
+import os
+import subprocess
+import shlex
+import urllib
+import time
+
+import flask
+from flask import Blueprint
+from flask import render_template
+from flask import request
+from flask import current_app
+from flask import stream_with_context
+
+from tools.sense_studio import utils
+
+train_bp = Blueprint('train_bp', __name__)
+
+PROCESS = None
+
+
+@train_bp.route('/<string:project>', methods=['GET'])
+def training_page(project):
+    project = urllib.parse.unquote(project)
+    return render_template('train.html', project=project, models=utils.BACKBONE_MODELS, is_disabled=False)
+
+
+def stream_template(template_name, **context):
+    # Ref: https://flask.palletsprojects.com/en/1.1.x/patterns/streaming/
+    current_app.update_template_context(context)
+    t = current_app.jinja_env.get_template(template_name)
+    rv = t.stream(context)
+    return rv
```

Comment:
It looks like because of this streaming mechanism, the footer of the page is only rendered when the training is done. I'm not sure yet, if this streaming functionality is the best solution for our case, because it is originally intended for splitting up an "enormous amount of data", not for updating delayed information.

Reply:
I'll investigate more on `multiprocessing` and `threading` if we can do that

Reply:
Here is an example using `EventSource`, which I think should work well for us: https://stackoverflow.com/a/13388915/6611214

Reply:
I referred the same link and you can see that after this `EventSource` approach the user  has  also posted `stream_template` approach.

Reply:
Yes, but is there an advantage to the `stream_template` approach?

Reply:
I have not tried `EventSource` in the first place and `stream_template` is provided by Flask itself. That's why I thought of using it as flask can handle its own thing properly. But there is no problem in investigating new thing which makes the feature as we want.